netsuite/client.py

```diff
@@ -83,8 +84,59 @@ def wrapper(self, *args, **kw):
     return decorator
 
 
+class NetSuiteTransport(Transport):
+    """
+    NetSuite dynamic domain wrapper for zeep.transports.transport
+
+    Latest NetSuite WSDL now uses relative definition addresses
+
+    zeep maps reflective remote calls to the base WSDL address,
+    rather than the dynamic subscriber domain
+
+    Wrap the zeep transports service with our address modifications
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Assign the dynamic host domain component to a class variable
+        """
+
+        _wsdl = kwargs['wsdl_url']
+
+        kwargs.pop('wsdl_url',None)
+        
+        # Find first slash after domain name
+        idx = _wsdl.index('/', 8) + 1
+
+        # We only want the dynamic domain
+        _wsdl=_wsdl[:idx]
+        self._wsdl = _wsdl
+        
+        super(NetSuiteTransport, self).__init__(**kwargs)
```

Comment:
I believe in Python 3 this will throw an error and should maybe be:

```python
super().__init__(**kwargs)
```

Reply:
When run locally, this leads to this error:

```
Traceback (most recent call last):
  File "foo.py", line 90, in <module>
    nstOG = NetSuiteTransportOG(wsdl_url=wsdl_url)
  File "foo.py", line 81, in __init__
    super(NetSuiteTransport, self).__init__(**kwargs)
TypeError: super(type, obj): obj must be an instance or subtype of type
```

Reply:
Regarding python 3 compatibility, I have run a version of this code in P3 with no issues.  This structure was provided by a Python 2 developer, which I believe that this code should still support, especially if we are going to create a pull request back to the project owner.

I will examine the error condition report.

Reply:
The original package only lists Python 3.x support. A few places this is demonstrated:

- The little graphic near the header at https://github.com/jmagnusson/netsuite
- Tests which only test for 3.x at https://github.com/jmagnusson/netsuite/blob/master/.travis.yml
- The package information with only lists 3.x at https://github.com/jmagnusson/netsuite/blob/master/setup.py#L36

_(We also have a PSWD suggestion that all new code be Python 3: https://confluence.merit.edu/display/SD/Python)_